pkg/webservice/handlers.go

```diff
@@ -235,3 +236,57 @@ func getNodeJSON(nodeInfo *cache.NodeInfo) *dao.NodeDAOInfo {
 		Schedulable: nodeInfo.IsSchedulable(),
 	}
 }
+
+func GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+
+	var result []*dao.ApplicationHistoryDAOInfo
+	lists := gClusterInfo.ListPartitions()
+	for _, k := range lists {
```

Comment:
right now it reports back per partition
I don't think front-end expect this. we should aggregate this to the cluster level. Can we change the `HistoricPartitionInfo` to `HistoricClusterInfo`?

Reply:
Agreed. Modified this and other classes in the PR accordingly.